.tekton/agent-helm-build-pull-request.yaml

```
@@ -0,0 +1,537 @@
+apiVersion: tekton.dev/v1
+kind: PipelineRun
+metadata:
+  annotations:
+    build.appstudio.openshift.io/repo: https://github.com/rh-gitops-midstream/release?rev={{revision}}
+    build.appstudio.redhat.com/commit_sha: '{{revision}}'
+    build.appstudio.redhat.com/pull_request_number: '{{pull_request_number}}'
+    build.appstudio.redhat.com/target_branch: '{{target_branch}}'
+    pipelinesascode.tekton.dev/cancel-in-progress: "true"
+    pipelinesascode.tekton.dev/max-keep-runs: "3"
+    pipelinesascode.tekton.dev/on-cel-expression: event == "pull_request" && target_branch
+      == "main" && 
+      ( 
+        "sources/argocd-agent/***".pathChanged() ||
+        ".tekton/agent-helm-build-pull-request.yaml".pathChanged()
+       )
+  creationTimestamp: null
+  labels:
+    appstudio.openshift.io/application: test-argocd-agent-helmchart
+    appstudio.openshift.io/component: agent-helm-build
+    pipelines.appstudio.openshift.io/type: build
+  name: agent-helm-build-on-pull-request
```

```suggestion
    appstudio.openshift.io/application: argocd-agent-helm-chart-main
    appstudio.openshift.io/component: argocd-agent-helm-chart
    pipelines.appstudio.openshift.io/type: build
  name: argocd-agent-helm-chart-on-pull-request
```
Nits to maintain consistent naming across components.